src/main/kotlin/com/ampnet/userservice/grpc/GrpcUserServer.kt

```diff
@@ -133,12 +140,41 @@ class GrpcUserServer(
         responseObserver.onCompleted()
     }
 
-    private fun getRole(role: SetRoleRequest.Role): UserRole =
+    override fun getAllActiveUsers(
+        request: PlatformManagerRequest,
+        responseObserver: StreamObserver<UsersWithExtendedInfoResponse>
+    ) {
+        logger.debug { "Received gRPC request getAllActiveUsers for coop: ${request.coop}" }
+        try {
+            validateUserIsPlatformManager(request)
+            val users = userRepository.findAllByCoopAndUserInfoUuidIsNotNull(request.coop)
+                .associateBy { it.userInfoUuid }
+            val usersWithExtendedInfo = userInfoRepository.findAllByCoop(request.coop).map { userInfo ->
+                users[userInfo.uuid]?.let { user -> buildUserWithExtendedInfoResponse(user, userInfo) }
+            }
+            val coop = coopService.getCoopByIdentifier(request.coop)
+                ?: throw ResourceNotFoundException(ErrorCode.COOP_MISSING, "Missing coop: ${request.coop} on platform")
```

Comment:
Fetch coop before fetching users. It's a bit cleaner flow, users must belong to the coop and the SQL request is simpler.